### DISCUSSION TITLE
Date and time should be displayed rather than an approximation such as "last week" or "today"

### DISCUSSION THREAD

**Discussion by jlswaugh** [community/community#122848] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Oftentimes, the exact date and time that a commit is made is important. The exact date and time should be specified, rather than a relatively unhelpful approximation such as "last week" or "today".

**Comment by hafiz-muhammad** [comment]


**Reply by RobertKrawitz** [reply]
That requires a lot of work for something really vital.  Relative or "friendly" (i. e. colloquial) dates simply aren't useful for this purpose (personally, I'd prefer absolute dates for everything, categorically).

There's a lot of other discussion of this topic that really needs to be addressed; see e. g. https://github.com/orgs/community/discussions/5972

**Comment by dchmelik** [comment]
Yes; the current way it displays is useless/garbage/hassle.  I check projects for updates most days (unless busy which can happen for weeks) and save them in format software-date-hash.tar.gz (xz would be preferable) with date in format YYYYMMDD though on project pages it's best displayed as ISO time (YYYY-MM-DD followed by a hyphen then 24-hour time).  If it just says last week or last month then I don't know if I have the latest version.

**Comment by ahayes** [comment]
> ### Select Topic Area
> 
> Product Feedback
> ### Body
> 
> Oftentimes, the exact date and time that a commit is made is important. The exact date and time should be specified, rather than a relatively unhelpful approximation such as "last week" or "today".

Thank you for creating this feedback issue. I am honestly amazed it hasn't been dealt with already.

GitHub folks, please use [ISO 8601 datetime](https://www.iso.org/iso-8601-date-and-time-format.html). It is also described in a [W3C note](https://www.w3.org/TR/NOTE-datetime) and on [Wikipedia](https://en.wikipedia.org/wiki/ISO_8601) if you can't get approvals to buy the ISO specification document. You can always put an option in user's Appearance settings if you really want to keep the vague "x units ago" descriptions around as the default.

Here is how GitLab handles it in user preferences:

<img width="542" height="362" alt="Screenshot From 2026-06-16 17-02-36" src="https://github.com/user-attachments/assets/9e236a99-b733-4d1f-a0a5-dded4b0dea4f" />

**Comment by candsastle** [comment]
This issue has frustrated me quite a number of times. It would really be nice if the specific datetimes were given.
